src/common/utils/time.ts

```diff
@@ -26,5 +26,5 @@ export function formatTime(seconds: number): string {
 }
 
 export function formatTime_(nHours: number, nMinutes: number, nSeconds: number): string {
-    return `${nHours > 0 ? (nHours.toString().padStart(2, "0") + ":") : ``}${nMinutes > 0 ? (nMinutes.toString().padStart(2, "0") + ":") : `00:`}${nSeconds > 0 ? (nSeconds.toString().padStart(2, "0")) : `00`}`;
+    return `${nHours > 0 ? (nHours.toString().padStart(2, "0") + ":") : ``}${nMinutes > 0 ? (nMinutes.toString().padStart(2, "0") + ":") : `00:`}${nSeconds > 0 ? (nSeconds.toString(3).padStart(2, "0")) : `00`}`;
```

Comment:
Are you sure about changing the `.toString(radix)` to 3?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString
`node -e "console.log((123.45).toString())"` => `123.45`
`node -e "console.log((123.45).toString(3))"` => `11120.11001100110011001100110011002`

Reply:
Fun fact: `(60).toString(3)` is `2020` ;)

Reply:
If you want to display milliseconds, you could of course use the fractional representation of the `nSeconds` number itself (because base 10), but to be consistent I would suggest passing an additional parameter `nMilliseconds: number` (which really is an integer between `000` and `999` inclusive ... but we have to use `number` just like with the other function arguments, for example `nSeconds` really is `[00, 60[`).
Then, add to the constructed string: `.${nMilliseconds > 0 ? (nMilliseconds.toString().padStart(3, "0")) : `000`}` (note the `.` dot prefix, and the 3-zero-padding)

Reply:
you're right

it's pretty bad